src/plugins/visualizations/public/visualize_app/components/visualize_listing.tsx

```diff
@@ -166,16 +172,84 @@ export const VisualizeListing = () => {
         listingLimit,
         references,
         referencesToExclude
-      ).then(({ total, hits }: { total: number; hits: Array<Record<string, unknown>> }) => ({
-        total,
-        hits: hits
+      ).then(({ total, hits }: { total: number; hits: Array<Record<string, unknown>> }) => {
+        const content = hits
           .filter((result: any) => isLabsEnabled || result.type?.stage !== 'experimental')
-          .map(toTableListViewSavedObject),
-      }));
+          .map(toTableListViewSavedObject);
+
+        visualizedUserContent.current = content;
+
+        return {
+          total,
+          hits: content,
+        };
+      });
     },
     [listingLimit, uiSettings, savedObjects.client]
   );
 
+  const onInspectorSave = useCallback(
+    async (args: { id: string; title: string; description?: string; tags: string[] }) => {
+      const content = visualizedUserContent.current?.find(({ id }) => id === args.id);
+
+      if (content) {
+        await updateBasicSoAttributes(
+          content.id,
+          content.type,
+          {
+            title: args.title,
+            description: args.description ?? '',
+            tags: args.tags,
+          },
+          { savedObjectsClient: savedObjects.client, overlays, savedObjectsTagging }
+        );
+      }
+    },
+    [overlays, savedObjects.client, savedObjectsTagging]
+  );
+
+  const inspectorValidators: OpenInspectorParams['customValidators'] = useMemo(
+    () => ({
+      title: [
+        {
+          type: 'warning',
+          async fn(value, id) {
+            if (id) {
+              const content = visualizedUserContent.current?.find((c) => c.id === id);
+              if (content) {
+                try {
+                  await checkForDuplicateTitle(
+                    {
+                      id,
+                      title: value,
+                      lastSavedTitle: content.title,
+                      getEsType: () => content.type,
+                    },
+                    false,
+                    false,
+                    () => {},
+                    { savedObjectsClient: savedObjects.client, overlays }
+                  );
+                } catch (e) {
+                  return i18n.translate(
```

Comment:
Sorry for the late comment. Seeing this PR after some PTO. 
Is it possible that `checkForDuplicateTitle` would throw for any other reason than a duplicate title? Server error, js error? If so we would probably want to send a specific error code that we could read here.

Discard this comment if `checkForDuplicateTitle` does handle system errors gracefully.